File: Python/Array_101/1_Intro/max_consecutive_ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        ans = 0
        temp = 0
        con_flag = False
        for i in nums:
            if i == 0:
                if ans < temp:
                    ans = temp
                temp = 0
            else:
                temp += 1
        if ans < temp:
            ans = temp
        return ans

File: Java/DataStructures/Array_101/1_Intro/MaxConsecutiveOnes.java
class MaxConsecutiveOnes {
    public int findMaxConsecutiveOnes(int[] nums) {
        int n = nums.length;
        int max = 0;
        int t = nums[0];
        for (int i = 1; i < n; i++) {
            if (nums[i] == 0) {
                max = max < t ? t : max;
                t = 0;
            } else
                t += 1;
        }
        return max < t ? t : max;
    }
}